api/testdata/test.yaml

```diff
@@ -5,7 +5,11 @@ metadata:
 spec:
   packageName: Test
   packageVersion: v0.1.0
-  config:
-    config.foo: windom
-    secret.bar: eolus
-    pvc.storage: 1Mi
+  config: |
+    config:
+      - foo: windom
+      - baz: baz
+    secret:
+      bar: eolus
+    pvc:
+      storage: 1Mi
```

Comment:
I haven't looked at the code in general, but I like the general idea of this although I think we should attempt to keep the yaml flat. The helm best practices  of keeping values.yaml flat as possible I think also would help our situation

Reply:
We'd have to try to enforce that on each chart we produce. In general most charts keep it only 1-3 levels deep.